### DISCUSSION TITLE
What is the difference between ID and databaseID?

### DISCUSSION THREAD

**Discussion by mljrg** [community/community#24855] [Apps, API and Webhooks]
<p>Hello,</p>
<p>I am starting a new integration with GitHub API v4 (the one using GraphQL).</p>
<p>I see for example that the User and Repository objects have the fields <code>id</code> and <code>databaseid</code>.</p>
<p>What is the difference between them?</p>
<p>Which should I use?</p>
<p>Thanks</p>

**Comment by mljrg** [comment]
<p>Hi,</p>
<p>so many views and no one can help on this question ?!?</p>
<p> ¯_(ツ)_/¯</p>

**Reply by coffeeandchill229** [reply]
https://docs.github.com/en/graphql/guides/using-global-node-ids
using id should be good

**Comment by aoi92** [comment, marked as answer]
<table><tbody>
<tr>
<td><b>type</b></td>
<td><b>api v3</b></td>
<td><b>api v4</b></td>
<td><b>Exemple</b></td>
</tr>
<tr>
<td>int (8)</td>
<td>id</td>
<td>databaseId</td>
<td>12345678</td>
</tr>
<tr>
<td>String (20)</td>
<td>node_id</td>
<td>id</td>
<td>MD96VXZlcjU0Mpd4NTY0</td>
</tr>
</tbody></table>On API v4, you should use **id** which is the main id for graphQL.

**Reply by nvmnghia** [reply]
I think you can use `databaseId` to query in REST, but it's unreliable e.g. sometimes it works sometimes not.
